app/src/main/java/com/github/polybooks/core/database/SalesAdapter.kt

```diff
@@ -14,20 +15,55 @@ import com.github.polybooks.core.Sale
  */
 class SalesAdapter(var salesList: List<Sale>): RecyclerView.Adapter<SalesAdapter.SalesViewHolder>() {
     class SalesViewHolder(itemView : View) : RecyclerView.ViewHolder(itemView){
-        val mName: TextView = itemView.findViewById(R.id.BookName)
-        val mPrice : TextView = itemView.findViewById(R.id.SalePrice)
+        val viewTitle: TextView = itemView.findViewById(R.id.text_view_title)
+        val viewEdition: TextView = itemView.findViewById(R.id.text_view_edition)
+        val viewBy: TextView = itemView.findViewById(R.id.text_view_by)
+        val viewAuthor: TextView = itemView.findViewById(R.id.text_view_author)
+        val viewCondition: TextView = itemView.findViewById(R.id.text_view_condition)
+        val viewPrice: TextView = itemView.findViewById(R.id.text_view_price)
     }
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SalesViewHolder {
-        val v : View = LayoutInflater.from(parent.context).inflate(R.layout.sale_item, parent, false)
-        return SalesAdapter.SalesViewHolder(v)
+        val v: View = LayoutInflater.from(parent.context).inflate(R.layout.sale_item, parent, false)
+        return SalesViewHolder(v)
     }
 
     override fun onBindViewHolder(holder: SalesViewHolder, position: Int) {
-        val currentSale : Sale = salesList[position]
 
-        holder.mName.setText(currentSale.book.title)
-        holder.mPrice.setText( String.format("%.2f",currentSale.price))
+        /**
+         * Creates an authors string from a list of author(s)
+         */
+        fun authorsFromList(list: List<String>): String {
+            fun authorsFromListRec(acc: String, list: List<String>): String {
+                return if (list.size == 1) "${acc}and ${list[0]}"
+                else authorsFromListRec("${list[0]}, ", list.drop(1))
+            }
+
+            if (list.size == 1) return list[0]
+            else return authorsFromListRec("", list)
+        }
```

Comment:
Can never happen since the field "author" is deleted otherwise. But I fully agree! I added this case in the function and also simplified its call a bit!